R/peaks-functions.R

```diff
@@ -429,3 +429,65 @@ joinPeaksGnps <- function(x, y, xPrecursorMz = NA_real_,
     x[, 2L] <- MsCoreUtils::smooth(x[, 2L], cf = coef)
     x
 }
+
+#' @description
+#'
+#' Removes (Orbitrap) FFT artefacts (peaks) from a spectrum keeping peaks
+#' potentially representing [13]C isotopes.
+#'
+#' @param x peaks matrix
+#'
+#' @param halfWindowSize `numeric(1)` defining the (m/z) window left and right
+#'     of a peak to look for artefacts.
+#'
+#' @param threshold `numeric(1)` defining the relative intensity to the actual
+#'     peak below which peaks are considered artefacts. Defaults to
+#'     `threshold = 0.2` hence removing all peaks with an intensity below 0.2 *
+#'     the peak intensity.
+#'
+#' @param keepIsotopes `logical(1)` whether it should be screened for potential
+#'     isotope peaks. Defaults to `keepIsotopes = TRUE` thus candidate artefact
+#'     peaks potentially being isotopes will **not** be removed.
+#'
+#' @param maxCharge `integer(1)`
+#'
+#' @param isotopeTolerance `numeric(1)`
+#'
+#' @author Jan Stanstrup
+#'
+#' @importFrom MsCoreUtils between
+#'
+#' @noRd
+.peaks_remove_fft_artifact <- function(x, halfWindowSize = 0.05,
+                                       threshold = 0.2,
+                                       keepIsotopes = TRUE,
+                                       maxCharge = 5,
+                                       isotopeTolerance = 0.005) {
+    neutron   <- 1.0033548378 # really C12, C13 difference
```

Comment:
Would it be useful to make this an argument? (I don't know if anybody ever want's to change this.)

Reply:
The only scenario I can think of where you want to change this is if you run an orbitrap in super duper high resolution mode so that there is isotopic fine-pattern. In that case though you'd need to define a vector of possible distances and not a single number...

Reply:
In the longer run we could have here also other isotopes (not only 13C, but also combinations etc). But for now I would keep it like this.